### PR TITLE
Fix visibility on addSessionAuthenticationStrategy in SessionManagementConfigurer

### PR DESCRIPTION
**Pull request by jebeaudet**
Hi

I believe this is a simple oversight since the class is `final` and the documentation points toward a public facing method.

A backport would be appreciated if possible!

Thanks

### CODE REVIEW

**Comment by sjohnr** [comment]
Hi @jebeaudet, thanks for the patch. Before proceeding, could you explain a bit more about the issue you're facing and how you hope to solve it with this change?

From looking at the code, it looks like the design of the `SessionManagementConfigurer` does not intend for applications to use this method directly. However, it appears there may be a workaround through the `CsrfConfigurer` if you're interested in doing this in the meantime.

**Comment by jebeaudet** [comment]
Oh ok yea I see that it's used in a package protected fashion.

I had a use case where I wanted to add an audit on authentication via a session (not only for new authentication which is handled by the `ApplicationEventPublisher`). But after some tests I realized that this was not the proper place to do so anyway so I added a filter before `SessionManagementFilter`. 

I opened this PR anyway because I thought this was a simple oversight but now that you mention `CsrfConfigurer`, I realize this might have been intended but the intent is not clear as to why. IMO, it could be a valid use case to do something special with the `SessionAuthenticationStrategy` but I don't have a valid use case on top of my head.

TL;DR, feel free to close if you want, thanks!

**Comment by sjohnr** [comment]
@jebeaudet thanks!

> IMO, it could be a valid use case to do something special with the SessionAuthenticationStrategy but I don't have a valid use case on top of my head.

It very well could be, though there could also be other approaches. This does sound vaguely familiar though, and I wonder if it's related to the idea of intercepting the result of AuthenticationProvider#authenticate, mentioned in [this comment](https://github.com/spring-projects/spring-security/issues/10939#issuecomment-1064721373) of #10939.

In any case, feel free to open another issue if there's something you feel isn't possible or add a comment to the above issue if it's relevant to you. I'll go ahead and close this for now as I'm not sure it would be desired to open up the method in this PR.
